src/components/form/ContourForm.tsx

```diff
@@ -0,0 +1,242 @@
+import React, { useEffect } from 'react';
+import {
+  Form,
+  FormGroup,
+  TextInput,
+  Switch,
+  Button,
+  Alert,
+} from '@patternfly/react-core';
+import { useParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+
+import { ContourFormProps, Route } from '../../types';
+import { useK8sResources } from '../../hooks/useK8sResources';
+import RouteForm from './RouteForm';
+import CustomDropdown from '../shared/CustomDropdown';
+import { DEFAULT_ROUTE } from '../../constants';
+
+const ContourForm = ({
+  formData,
+  updateFormData,
+  onChange,
+  isEdit = false,
+  validationErrors = [],
+  saveError = null,
+  onSubmit,
+}: ContourFormProps) => {
+  const { t } = useTranslation('plugin__contour-console-plugin');
+  const { ns: namespace } = useParams<{ ns: string; name?: string }>();
+  const { services, secrets, ingressClasses } = useK8sResources(namespace);
+
+  useEffect(() => {
+    onChange(formData);
+  }, [formData]);
```

Comment:
You can remove this useEffect based on this description:
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes